### DISCUSSION TITLE
Repeated connection request after disconnect

### DISCUSSION THREAD

**Discussion by voenkomatiwe** [WalletConnect/walletconnect-monorepo#4205] [Developer Support]
Hey everyone.
I'm developing an application in which you can add several wallets. The action should be the following:
The user connects the wallet and signs the message. I encountered this problem:
- I make a disconnect (from wagmi);
- Open a modal window;
- I click on metamask, and immediately connect to the wallet to which I was already connected.

Is it possible to somehow make a repeated request to connect an account? Example of this method `wallet_requestPermissions`

**Comment by glitch-txs** [comment]
you would need to customize the injected provider but not sure if all extension wallets support `wallet_requestPermissions`

**Reply by glitch-txs** [reply]
notice that for MetaMask we use it's EIP-6963 provider, wagmi wraps it from mipd into an injected connector

**Reply by voenkomatiwe** [reply]
I noticed that Vagmi suggests using 2 more options for ready-made solutions with connecting wallets:
- [ConnectKit](https://docs.family.co/connectkit/try-it-out)
- [RainbowKit](https://www.rainbowkit.com/)

And when I again connect my wallet (Metamask), they ask as if it’s my first time here.

Is it possible to do the same on [web3modal](https://web3modal.com/)?

**Reply by glitch-txs** [reply]
Is that Wagmi v2?

**Reply by voenkomatiwe** [reply]
most likely yes, but on version wagmi v1 it works the same way

**Reply by Soniachniy** [reply]
@glitch-txs it seems that connect-kit just ask wallet_requestPermissions before connection just to be sure that user connect the right wallet, https://eips.ethereum.org/EIPS/eip-2255 is basic interface that should be on other wallets, not only on MM. Is there any way to do that in the nearest future or it's not relevant at all?

**Reply by voenkomatiwe** [reply]
@glitch-txs  Thank you very much for your answers. This discussion is no longer required.

**Reply by glitch-txs** [reply]
@Solniechniy feel free to open a new discussion, but this is handled by Wagmi internally
